src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1802,58 +1804,85 @@ private void visitEnumType(String symbolName, String qualifiedName, EnumType typ
         return;
       }
 
-      boolean isNumericEnum = type.getElementsType().getPrimitiveType().equals(numberType);
-      if (isNumericEnum) {
-        emit("enum");
+      JSType primitiveType = type.getElementsType().getPrimitiveType();
+      if (maybeStringOrNumericEnum(primitiveType, unqualifiedName, node)) {
+        return;
+      }
+
+      visitTypeAlias(type.getElementsType().getPrimitiveType(), unqualifiedName, true);
+      emit("var");
+      emit(unqualifiedName);
+      emit(": {");
+      emitBreak();
+      indent();
+      for (String elem : sorted(type.getElements())) {
+        emit(elem);
+        emit(":");
+        // No need to use type.getMembersType(), this must match the type alias we just declared.
         emit(unqualifiedName);
-        emit("{");
+        emit(",");
         emitBreak();
-        indent();
+      }
+      unindent();
+      emit("}");
+      emitNoSpace(";");
+      emitBreak();
+    }
+
+    /**
+     * Attempt to convert the node to a string enum or numeric enum and return a boolean indicating
+     * whether it successfully emitted the enum.
+     */
+    private boolean maybeStringOrNumericEnum(
+        JSType primitiveType, String unqualifiedName, Node node) {
+      if (!primitiveType.equals(numberType) && !primitiveType.equals(stringType)) {
+        return false;
+      }
 
-        // The current node points to the enum type declaration, this means that the next node will
-        // be the OBJECTLIT containing all enum key and value pairs. However, globally declared
-        // enums that are indirectly provided will instead be pointing to the parent of the
-        // OBJECTLIT parent.
-        Stream<Node> elementStream =
-            node.getNext() != null
-                ? Streams.stream(node.getNext().children())
-                : Streams.stream(node.getFirstChild().children());
-        Map<String, Node> elements =
-            elementStream.collect(Collectors.toMap(Node::getString, Node::getFirstChild));
-
-        for (String elem : sorted(elements.keySet())) {
-          emit(elem);
-          @Nullable Node n = elements.get(elem);
-          if (n != null && n.isNumber()) {
+      // The current node points to the enum type declaration, this means that the next node will
+      // be the OBJECTLIT containing all enum key and value pairs. However, globally declared
+      // enums that are indirectly provided will instead be pointing to the parent of the
+      // OBJECTLIT parent.
+      Node objectOfAllMembers = node.getNext() != null ? node.getNext() : node.getFirstChild();
+
+      // Peek an enum member. If the Closure string enum's value is not literal don't emit anything
+      // and fall back to the safe conversion.
```

Comment:
what if the n-th member is non-literal?

```
{ A: 'foo', B: 'bar', C: f() }
```
I think you need to check them all.